examples/AdvancedOperations/AddDynamicSearchAds.php

```diff
@@ -0,0 +1,347 @@
+        printf("Added ad group named '%s'.%s", $adGroupResourceName, PHP_EOL);
+
+        return $adGroupResourceName;
+    }
+
+    /**
+     * Creates an expanded dynamic search ad.
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param string $customerId the client customer ID without hyphens
+     * @param string $adGroupResourceName the ad group resource name
+     */
+    private static function createExpandedDSA(
+        GoogleAdsClient $googleAdsClient,
+        string $customerId,
+        string $adGroupResourceName
+    ) {
+        $adGroupAd = new AdGroupAd([
+            'ad_group' => $adGroupResourceName,
+            'status' => AdGroupAdStatus::PAUSED,
+            'ad' => new Ad([
+                'expanded_dynamic_search_ad' => new ExpandedDynamicSearchAdInfo([
+                    'description' => new StringValue(['value' => 'Buy tickets now!'])
+                ])
+            ])
+        ]);
+
+        $adGroupAdOperation = new AdGroupAdOperation();
+        $adGroupAdOperation->setCreate($adGroupAd);
+
+        // Issues a mutate request to add the ad group ads.
+        $adGroupAdServiceClient = $googleAdsClient->getAdGroupAdServiceClient();
+        /** @var MutateAdGroupAdsResponse $adGroupAdResponse */
+        $adGroupAdResponse = $adGroupAdServiceClient->mutateAdGroupAds(
+            $customerId,
+            [$adGroupAdOperation]
+        );
+
+        $adGroupAdResourceName = $adGroupAdResponse->getResults()[0]->getResourceName();
+        printf("Added ad group ad named '%s'.%s", $adGroupAdResourceName, PHP_EOL);
+
+        return $adGroupAdResourceName;
+    }
+
+    /**
+     * Creates webpage targeting criteria for the DSA.
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param string $customerId the client customer ID without hyphens
+     * @param AdGroup $adGroup the ad group
+     */
+    private static function createWebPageCriteria(
+        GoogleAdsClient $googleAdsClient,
+        string $customerId,
+        string $adGroupResourceName
+    ) {
+        $adGroupCriterion = new AdGroupCriterion([
+            'ad_group' => new StringValue(['value' => $adGroupResourceName]),
+            'status' => AdGroupCriterionStatus::PAUSED,
+            'cpc_bid_micros' => new Int64Value(['value' => 10000000]),
+            'webpage' => new WebPageInfo([
+                'criterion_name' => new StringValue(['value' => 'Special Offers']),
+                'conditions' => [
+                    new WebPageConditionInfo([
+                        'operand' => WebPageConditionOperand::URL,
+                        'argument' => new StringValue(['value' => '/specialoffers'])
+                    ]),
+                    new WebPageConditionInfo([
+                        'operand' => WebPageConditionOperand::PAGE_TITLE,
+                        'argument' => new StringValue(['value' => 'Special Offer'])
```

Comment:
Nit: Is this intentionally made different than the criterion name (Offer vs Offers)?

Reply:
Fixed, thanks for spotting this!